$NetBSD: patch-aa,v 1.4 1999/03/31 10:05:57 agc Exp $

NetBSD patches for ELF/a.out calculation
Solaris patches to mimic NetBSD ones, for use in pkgsrc

--- ltconfig.orig	Fri Mar 20 09:00:29 1998
+++ ltconfig	Sun Feb 14 17:49:16 1999
@@ -808,7 +808,12 @@
 
   netbsd*)
     # Tested with NetBSD 1.2 ld
-    archive_cmds='$LD -Bshareable -o $lib$libobjs'
+    if $CC -E - -dM </dev/null | grep -q __ELF__; then
+      archive_cmds='$LD -shared -o $lib$libobjs'	# ELF
+    else
+      archive_cmds='$LD -Bshareable -o $lib$libobjs'	# a.out
+      cpprt0=/usr/lib/c++rt0.o
+    fi
     hardcode_libdir_flag_spec='-R$libdir'
     hardcode_direct=yes
     hardcode_shlibpath_var=no
@@ -1181,8 +1186,15 @@
 
 netbsd* | openbsd*)
   version_type=sunos
-  library_names_spec='${libname}${release}.so.$versuffix'
-  finish_cmds='PATH="$PATH:/sbin" ldconfig -m $libdir'
+  if $CC -E - -dM </dev/null | grep -q __ELF__; then
+    library_names_spec='${libname}${release}.so.$versuffix ${libname}${release}.so.$major ${libname}${release}.so'
+    soname_spec='${libname}${release}.so.$major'
+    dynamic_linker='NetBSD ld.elf_so'
+  else
+    library_names_spec='${libname}${release}.so.$versuffix'
+    finish_cmds='PATH="$PATH:/sbin" ldconfig -m $libdir'
+    dynamic_linker='NetBSD/OpenBSD (a.out) ld.so'
+  fi
   shlibpath_var=LD_LIBRARY_PATH
   ;;
 
@@ -1208,8 +1220,8 @@
   ;;
 
 solaris2*)
-  version_type=linux
-  library_names_spec='${libname}${release}.so.$versuffix ${libname}${release}.so.$major $libname.so'
+  version_type=sunos
+  library_names_spec='${libname}${release}.so.$versuffix ${libname}${release}.so.$major ${libname}${release}.so'
   soname_spec='${libname}${release}.so.$major'
   shlibpath_var=LD_LIBRARY_PATH
   ;;
@@ -1482,6 +1494,9 @@
 # Set to yes if using SHLIBPATH_VAR=DIR during linking hardcodes DIR into
 # the resulting binary.
 hardcode_shlibpath_var=$hardcode_shlibpath_var
+
+# C++ startup code
+cpprt0="$cpprt0"
 
 EOF
 
